onnxruntime/core/session/IOBinding.cc

```diff
@@ -88,20 +84,25 @@ common::Status IOBinding::BindOutput(const std::string& name, OrtDevice device)
 }
 
 common::Status IOBinding::BindOutputImpl(const std::string& name, const OrtValue& ml_value, OrtDevice device) {
-  auto rc = Contains(output_names_, name);
-  if (rc.first) {
-    outputs_[rc.second] = ml_value;
-    outputs_device_info_[rc.second] = device;
-  } else {
+  ORT_ENFORCE(mapped_output_names_.size() == output_names_.size(), "Size mismatch(1):", mapped_output_names_.size(), "!=", output_names_.size());
+  auto it = mapped_output_names_.emplace(name, output_names_.size());
+  size_t index = it.first->second;
+  if (it.second) {
     output_names_.push_back(name);
     outputs_.push_back(ml_value);
     outputs_device_info_.push_back(device);
+  } else {
+    outputs_[index] = ml_value;
+    outputs_device_info_[index] = device;
   }
+  ORT_ENFORCE(mapped_output_names_.size() == output_names_.size(), "Size mismatch(2)", mapped_output_names_.size(), "!=", output_names_.size());
+  ORT_ENFORCE(mapped_output_names_.size() == outputs_device_info_.size(), "Size mismatch(3)", mapped_output_names_.size(), "!=", outputs_device_info_.size());
```

Comment:
If multiple `ORT_ENFORCE`s are needed, may we log a more descriptive error message instead of `(1)`, `(2)` and `(3)`?